### ISSUE TITLE
Null Dereference in tcd_malloc_decode_tile

### ISSUE THREAD

**Issue by Javantea** [uclouvain/openjpeg#657]
A crafted jp2 file can cause openjpeg-1.5.2 to crash with segmentation fault. It appears to be a null dereference. This was found using fuzzing. The fuzzer is now open source: https://www.altsci.com/jrsfuzz/ I am happy to provide samples if they would be helpful.

This may or may not be related to #603.

Backtrace:

```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7bcd352 in tcd_malloc_decode_tile (tcd=0x637c00, image=0x6288c0, cp=0x619140, tileno=0, cstr_info=0x0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/tcd.c:710
710                     tilec->x0 = int_ceildiv(tile->x0, image->comps[compno].dx);
(gdb) bt
#0  0x00007ffff7bcd352 in tcd_malloc_decode_tile (tcd=0x637c00, image=0x6288c0, cp=0x619140, tileno=0, cstr_info=0x0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/tcd.c:710
#1  0x00007ffff7bb683e in j2k_read_eoc (j2k=0x6190c0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/j2k.c:1691
#2  0x00007ffff7bb7062 in j2k_decode (j2k=0x6190c0, cio=0x6191e0, cstr_info=0x0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/j2k.c:2027
#3  0x00007ffff7bbaf0c in opj_jp2_decode (jp2=0x619050, cio=0x6191e0, cstr_info=0x0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/jp2.c:841
#4  0x00007ffff7bbe126 in opj_decode_with_info (dinfo=0x619010, cio=0x6191e0, cstr_info=0x0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/openjpeg.c:168
#5  0x00007ffff7bbe096 in opj_decode (dinfo=0x619010, cio=0x6191e0)
    at /jp2/openjpeg-1.5.2/libopenjpeg/openjpeg.c:157
#6  0x0000000000404342 in main (argc=5, argv=0x7fffffffd268)
    at /jp2/openjpeg-1.5.2/applications/codec/j2k_to_image.c:681

(gdb) list
705                             cp->tileno[tileno] = -1;
706                             return;
707                     }
708
709                     /* border of each tile component (global) */
710                     tilec->x0 = int_ceildiv(tile->x0, image->comps[compno].dx);
711                     tilec->y0 = int_ceildiv(tile->y0, image->comps[compno].dy);
712                     tilec->x1 = int_ceildiv(tile->x1, image->comps[compno].dx);
713                     tilec->y1 = int_ceildiv(tile->y1, image->comps[compno].dy);
714
(gdb) print tilec
$1 = (opj_tcd_tilecomp_t *) 0x0

(gdb) print tile
$2 = (opj_tcd_tile_t *) 0x642290
(gdb) print image
$3 = (opj_image_t *) 0x6288c0
(gdb) print image->comps
$4 = (opj_image_comp_t *) 0x628900
(gdb) print image->comps[compno]
$5 = {dx = 1, dy = 1, w = 1001, h = 2533, x0 = 0, y0 = 0, prec = 8, bpp = 0, sgnd = 0, resno_decoded = 5, factor = 0, data = 0x7ffff5bb4010}
(gdb) print tcd->tcd_image->tiles[0]
$11 = {x0 = 768, y0 = 2304, x1 = 1001, y1 = 2533, numcomps = 1, comps = 0x0, numpix = 0, distotile = 0, distolayer = {0 <repeats 100 times>}, 
  packno = 0}
(gdb) print tcd->tcd_image->tiles[1]
$12 = {x0 = 0, y0 = 0, x1 = 256, y1 = 256, numcomps = 1, comps = 0x0, numpix = 0, distotile = 0, distolayer = {0 <repeats 100 times>}, packno = 0}
```

**Comment by rouault** [comment]
Can you attach an image to reproduce the image ? (hopefully the issue is now fixed)

**Comment by Javantea** [comment]
[Repro image](https://www.altsci.com/jrsfuzz/repro/fuzzfile8_76ff.jp2)
More repro files exist if you're curious but that should help. Gentoo is still providing 1.5.2 and mplayer is still using it, so I'd like to verify against the 1.5.x releases. This bug doesn't affect 2.x series as far as I've tested. Should I take it up with them?

**Comment by rouault** [comment]
The project does no longer maintain the 1.5 series. There have been many security fixes since then, so people still using 1.5 should seriously consider migrating. Closing as fixed
